AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/SubscriptionSync/IncomingAsyncSubscriptionEventPublisher.swift

```diff
@@ -67,72 +68,81 @@ final class IncomingAsyncSubscriptionEventPublisher: AmplifyCancellable {
+        self.onUpdateOperation = await retryableOperation(
+            subscriptionType: .update,
+            modelSchema: modelSchema,
+            authModeStrategy: authModeStrategy,
+            auth: auth,
+            api: api
+        )
+        onUpdateOperation?.subscribe()
+            .sink(receiveCompletion: genericCompletionListenerHandler(result:), receiveValue: onUpdateValueListener!)
+            .store(in: &cancelables)
 
         // onDelete operation
-        let onDeleteValueListener = onDeleteValueListenerHandler(event:)
-        let onDeleteAuthTypeProvider = await authModeStrategy.authTypesFor(schema: modelSchema,
-                                                                           operations: [.delete, .read])
-        self.onDeleteValueListener = onDeleteValueListener
-        self.onDeleteOperation = RetryableGraphQLSubscriptionOperation(
-            requestFactory: IncomingAsyncSubscriptionEventPublisher.apiRequestFactoryFor(
-                for: modelSchema,
-                subscriptionType: .onDelete,
-                api: api,
-                auth: auth,
-                awsAuthService: self.awsAuthService,
-                authTypeProvider: onDeleteAuthTypeProvider),
-            maxRetries: onUpdateAuthTypeProvider.count,
-            resultListener: genericCompletionListenerHandler) { nextRequest, wrappedCompletion in
-            api.subscribe(request: nextRequest,
-                          valueListener: onDeleteValueListener,
-                          completionListener: wrappedCompletion)
-        }
-        onDeleteOperation?.main()
+        self.onDeleteValueListener = onDeleteValueListenerHandler(event:)
+        self.onDeleteOperation = await retryableOperation(
+            subscriptionType: .delete,
+            modelSchema: modelSchema,
+            authModeStrategy: authModeStrategy,
+            auth: auth,
+            api: api
+        )
+        onDeleteOperation?.subscribe()
+            .sink(receiveCompletion: genericCompletionListenerHandler(result:), receiveValue: onDeleteValueListener!)
+            .store(in: &cancelables)
+    }
+
+
+    func retryableOperation(
+        subscriptionType: IncomingAsyncSubscriptionType,
+        modelSchema: ModelSchema,
+        authModeStrategy: AuthModeStrategy,
+        auth: AuthCategoryBehavior?,
+        api: APICategoryGraphQLBehavior
+    ) async -> RetryableGraphQLSubscriptionOperation<Payload> {
+        let authTypeProvider = await authModeStrategy.authTypesFor(
+            schema: modelSchema,
+            operations: subscriptionType.operations
+        )
+
+        return RetryableGraphQLSubscriptionOperation(
+            requestStream: authTypeProvider.publisher()
+                .map { Optional.some($0) } // map to optional to have nil as element
+                .replaceEmpty(with: nil) // use a nil element to trigger default auth if no auth provided
```

Comment:
can we expand on this comment to explain in more detail? or do you prefer wrapping AWSAuthorizationType in something like another enum containing the default option?
```
enum AuthorizationTypeOrDefault {
  case inferDefault
  case authType(AWSAuthorizationType)
}
```

Reply:
yeah, as we've desicussed before, I will create a POC based on this solution.